### PR TITLE
[WIP] Consumer and producer pools

### PR DESCRIPTION
**Pull request by ianks**

### CODE REVIEW

File: multi_lookup.c [hunk 1]
```diff
@@ -49,37 +49,52 @@ void* read_file(void* filename)
   }
 
   fclose(file);
+  pthread_exit(NULL);
+
   return NULL;
 }
 
-int req_pool(char* filenames[])
+void* req_pool(void* files)
```
Comment (indiesquidge) [inline multi_lookup.c]: Why'd you change this?

Comment (ianks) [inline multi_lookup.c]: So now we have two 'main' threads: producer and consumer. In order to pass the `filenames` array we have to pass it as a `void*` buffer then cast it back to the correct type (`char**`).

This way, we get true asynchronous IO.

Comment (indiesquidge) [inline multi_lookup.c]: And why is the correct type now a double pointer?

Comment (ianks) [inline multi_lookup.c]: It was always that. `char**` is the same as saying `char* foo[]`

Comment (indiesquidge) [inline multi_lookup.c]: Oh that's right. It's just a pointer to an array. I always forget it's written like that.

Comment (ianks) [inline multi_lookup.c]: Yeah, I much prefer the `char* foo[]` syntax when possible but in this case we must cast it with `char**` AFAIK

File: ref/Mutex.c [hunk 2]
```diff
@@ -1,48 +0,0 @@
-
```
Comment (indiesquidge) [inline ref/Mutex.c:1]: Why get rid of this as a ref file?

File: ref/SimpleThread.c [hunk 3]
```diff
@@ -1,63 +0,0 @@
-
```
Comment (indiesquidge) [inline ref/SimpleThread.c:1]: And why get rid of this too?

File: ref/lookup.c [hunk 4]
```diff
@@ -61,13 +61,15 @@ main(int argc, char* argv[])
...
+      printf("%s %s %d\n", hostname, firstipstr, sizeof(firstipstr));
+
       if(dnslookup(hostname, firstipstr, sizeof(firstipstr)) == UTIL_FAILURE) {
         fprintf(stderr, "dnslookup error: %s\n", hostname);
         strncpy(firstipstr, "", sizeof(firstipstr));
       }
 
       /* Write to Output File */
-      fprintf(outputfp, "%s,%s\n", hostname, firstipstr);
+      /* fprintf(outputfp, "%s,%s\n", hostname, firstipstr); */
```
Comment (indiesquidge) [inline ref/lookup.c:72]: Why are we changing this file? It's supposed to be there as reference.

Comment (ianks) [inline ref/lookup.c:72]: You can change it back if you want.

**Pull request merged**